cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -824,8 +824,15 @@ class two_table_comparator {
 }  // namespace lexicographic
 
 namespace hash {
+
+template <template <typename> class hash_function, typename Nullate>
+class device_row_hasher;
+
+template <template <template <typename> class hash_function, typename> class device_row_hasher =
```

Comment:
I realize that you probably did this to reduce the number of lines of code that you had to change, but could we give a different name to the template class? The collision with the actual `device_row_hasher` caused me to take 5-10 mins longer than it should have to grok the point of this extra templating.

Reply:
I'd be happy to see those changes moved to a separate PR, but given time constraints I'm also OK with you tossing them into this one if you'd prefer.

Reply:
How about `DeviceRowHasher` for the template?

Reply:
Is there any way to just `template<typename Hasher>` instead of a long chain of `template` and `typename`? It takes me forever trying to guess what's going on.

Reply:
Nope. I tried it but template-template parameters can't be simplified like that. 😕

Reply:
Changed the template class name to `DeviceRowHasher` in 929c5890fff814a45ff7de520ec59f99779d7032.

Reply:
I'm fine with `DeviceRowHasher` for now. Anything to prevent the duplication.

To be more specific, the issue here is that `row_hasher::device_hasher`'s return value needs to be forwarded the template parameters from the function template. In order for that to work, we have to guarantee that the `DeviceRowHasher` type is a templated type, otherwise the compiler won't know that passing the template parameters along is legal. [This example demonstrates the issue (note the commented out line for the template class C).](https://godbolt.org/z/vTqnhfE6W)